src/Unused/Cache/Fingerprint.hs

```diff
@@ -0,0 +1,25 @@
+module Unused.Cache.Fingerprint
+    ( sha
+    ) where
+
+import System.Process
+
+sha :: IO String
+sha =
+    getSha <$> (fileList >>= md5Input >>= sortInput >>= md5Result)
+  where
+    getSha = head' . lines
+    head' (x:_) = x
+    head' _ = ""
+
+fileList :: IO String
+fileList = readProcess "find" [".", "-type", "f", "-not", "-path", "*/tmp/unused/*", "-print0"] ""
+
+md5Input :: String -> IO String
+md5Input = readProcess "xargs" ["-0", "-P", "1", "md5"]
```

Comment:
@pbrisbin cool - I'd tried that originally, was running into issues when things got more complicated, and switched. I also could've sworn I'd seen perf issues, but I can't seem to reproduce. I'll make the change!